api/task_manager_test.cc

```diff
@@ -0,0 +1,107 @@
+        int sequence_number = it != req->query_parameters.end() ? boost::lexical_cast<int>(it->second) : 0;
+        it = req->query_parameters.find("parent_id");
+        task_id parent_id = it != req->query_parameters.end() ? task_id{utils::UUID{it->second}} : task_id::create_null_id();
+        co_await tms.invoke_on(0, [id, name = "test", sequence_number, keyspace = std::move(keyspace), table = std::move(table), type = std::move(type), entity = std::move(entity), parent_id] (task_manager& tm) -> future<> {
+            auto& modules = tm.get_modules();
+            auto it = modules.find(name);
+            auto _id = id;
+            if (it != modules.end()) {
+                // TODO Deexie use make_task
+                auto task_impl_ptr = std::make_unique<test_task>(it->second, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity));
+                auto task = seastar::make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
+                task->start();
+            } else {
+                co_await seastar::coroutine::return_exception(std::runtime_error("module not found"));
+            }
+        });
+        co_return id.to_sstring();
+    });
+
+    tmt::unregister_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->query_parameters["task_id"]}};
+        sharded<task_manager>& tms = ctx.tm;
+        task_manager::foreign_task_ptr task = co_await task_manager::lookup_task_on_all_shards(tms, id);
+        co_await tms.invoke_on(task.get_owner_shard(), [name = "test", t = std::move(task), id] (auto& tm) mutable {
+            auto task = t.release();
+            task->finish();
+            task->unregister_task();
+        });
+        co_return json_void();
+    });
+
+    tmt::finish_test_task.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->param["task_id"]}};
+        sharded<task_manager>& tms = ctx.tm;
+        task_manager::foreign_task_ptr task = co_await task_manager::lookup_task_on_all_shards(tms, id);
+        co_await tms.invoke_on(task.get_owner_shard(), [name = "test", t = std::move(task), id] (auto& tm) mutable {
+            auto task = t.release();
+            task->finish();
+        });
+        co_return json_void();
+    });
+
+    tmt::finish_test_task_with_error.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        task_id id = task_id{utils::UUID{req->param["task_id"]}};
+        sharded<task_manager>& tms = ctx.tm;
+        task_manager::foreign_task_ptr task = co_await task_manager::lookup_task_on_all_shards(tms, id);
+        co_await tms.invoke_on(task.get_owner_shard(), [name = "test", t = std::move(task)] (auto& tm) mutable {
+            auto task = t.release();
+            task->finish_failed(std::make_exception_ptr(std::runtime_error("Test error")));
```

Comment:
If the error string will be given as an optional param to finish_task, it could be passed up there
and be verified by the test.